models/type/src/field-type/transformers.ts

```diff
@@ -0,0 +1,90 @@
+import { Reference } from '@commercetools-test-data/commons';
+import {
+  Transformer,
+  buildField,
+  buildFields,
+} from '@commercetools-test-data/core';
+import { faker } from '@faker-js/faker';
+import * as CustomFieldEnumValue from '../custom-field-enum-value';
+import { TCustomFieldEnumValue } from '../custom-field-enum-value';
+import * as CustomFieldLocalizedEnumValue from '../custom-field-localized-enum-value';
+import { TCustomFieldLocalizedEnumValue } from '../custom-field-localized-enum-value';
+import FieldType from './builder';
+import { typeNames, graphqlTypenameByFieldTypeName } from './constants';
+import { TFieldName, TFieldType, TFieldTypeGraphql } from './types';
+
+const list = <T>(fn: (index?: number) => T) =>
+  Array.from({ length: faker.number.int({ min: 1, max: 7 }) }, (_, index) =>
+    fn(index)
+  );
+
+const resolveCommonTransformations = <T extends TFieldType | TFieldTypeGraphql>(
+  fields: TFieldType
+): T => {
+  const commonFields = {
+    name: fields.name as TFieldName,
+    __typename: graphqlTypenameByFieldTypeName[fields.name],
+  } as T;
+
+  switch (fields.name) {
+    case 'Enum':
+      return {
+        ...commonFields,
+        values: buildFields(
+          (fields.values as TCustomFieldEnumValue[]) ||
+            list(CustomFieldEnumValue.random),
+          'rest'
```

Comment:
shouldn't this type `rest` be based on the caller? I mean, this `resolveCommonTransformations` is called by all 3 transformers below

Reply:
Updated here: 561d23675c46fbf0307cc2f8c0fa2696220ede83